### PR TITLE
Reorganizes packages to include more metadata

### LINKED ISSUE #9: Publish a list of available packages

### PR DESCRIPTION
**Pull request by kritzcreek**
I don't think we really have a good idea of how to create a "history" of package versions yet, so I propose we go with a simpler design, focus on solving two pressing issues for now, and solve the rest when we have more information.

The two issues being solved here:
- We need enough metadata to produce an informative package listing
- We need to make sure package names once "allocated" are not reused by someone else

The way this is solved is by adding a file per package to the `index/` directory which contains all the information we need to include it in a package set, but also additional information for the package listing. We must never _delete_ a file from this directory, but changing them is fine for now until we figure out how we want to preserve package history.

We then make the current package-set by adding or removing packages from `src/packages.dhall`. We can use this file to produce the package listing as it contains all the metadata.

The top-level `package-set.dhall` file then only picks the fields it needs to produce a package set from `src/packages.dhall`.

To generate the package listing we use `dhall-to-json` and feed it into a handlebars template to produce a static HTML file.

Package listing design

Fixes #9

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -34,3 +36,14 @@ jobs:
...
+        run: |
+          cd generate-package-list
+          npm install --only=production
+          npm run build ${{ github.ref }}
+      - name: Upload Package Listing
+        uses: JamesIves/github-pages-deploy-action@releases/v3
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BRANCH: gh-pages
+          FOLDER: generate-package-list/dist
```
Comment (nomeata) [inline .github/workflows/release.yml:49]: I like
```suggestion
          FOLDER: generate-package-list/dist
          SINGLE_COMMIT: yes
```
else the `gh-pages` branch just keeps increasing the size of the repo.

File: Makefile [hunk 2]
```diff
@@ -7,4 +7,4 @@ check-format:
 	dhall format --check --transitive package-set.dhall;
 	@echo checked dhall files are formatted
 ci: check-format
-	vessel verify --version 0.5.10
+	vessel verify --version 0.5.13
```
Comment (nomeata) [inline Makefile:10]: Slightly unrelated: Is this the only place where we pin the motoko version? I.e. it is not part of the package set metadat? Should it mabye?

**Review by nomeata — APPROVED** [review]
Nice stuff!

File: .github/workflows/release.yml (hunk 1 above)
Comment (kritzcreek) [reply .github/workflows/release.yml:49]: Sounds good, thanks!

File: Makefile (hunk 2 above)
Comment (kritzcreek) [reply Makefile:10]: It is, and I agree I'd like to add it to the package set somewhere, but I'm not sure if we need this to be a version range? It is part of the tag name for now, so at least you can tell from that.

**Pull request merged**